packages/shaderlab/src/shaders/shadingPBR/FragmentPBR.glsl

```diff
@@ -235,6 +246,26 @@ SurfaceData getSurfaceData(Varyings v, vec2 aoUV, bool isFrontFacing){
         surfaceData.anisotropicN = getAnisotropicBentNormal(surfaceData);
     #endif
 
+    //Iridescence
+    #ifdef MATERIAL_ENABLE_IRIDESCENCE
+        surfaceData.iridesceceFactor = material_IridescenceInfo.x;
+        surfaceData.iridesceceIor = material_IridescenceInfo.y;
+
+        float iridesceceThicknessMin = material_IridescenceInfo.z;
+        float iridesceceThicknessMax = material_IridescenceInfo.w;
+        #ifdef MATERIAL_HAS_IRIDESCENCE_THICKNESS_TEXTURE
+           vec3 iridescenceThicknessInfo = (texture2D( material_IridescenceThicknessTexture, uv)).rgb;
+           surfaceData.iridescenceThickness = mix(iridesceceThicknessMin, iridesceceThicknessMax, iridescenceThicknessInfo.g);
+        #else
+           surfaceData.iridescenceThickness = iridesceceThicknessMax;
+        #endif
+
+        #ifdef MATERIAL_HAS_IRIDESCENCE_TEXTURE
+           vec3 iridecenceIntensity = (texture2D( material_IridescenceTexture, uv)).rgb;
```

Comment:
intensity is a float variable;
`float iridecenceIntensity = texture2D( material_IridescenceTexture, uv).r;`